src/main/java/frc/robot/systems/EveryBotIntakeFSM.java

```diff
@@ -0,0 +1,340 @@
+				handleFlipCounterClockWiseState();
+				return flipMotor.getEncoder().getPosition() <= 0;
+			case IDLE_STOP:
+				handleIdleStopState();
+				return true;
+			case OUTTAKING   :
+				handleOuttakingState(null);
+				return true;
+			default:
+				throw new IllegalStateException("Invalid state: " + currentState.toString());
+		}
+	}
+
+	/*-------------------------NON HANDLER METHODS ------------------------- */
+	/**
+	 * Returns the type of object currently in the grabber.
+	 * @return int 0 1 or 2 for nothing, cone, cube
+	 */
+	public static ItemType getObjectType() {
+		return itemType;
+	}
+	/* ======================== Private methods ======================== */
+	/**
+	 * Decide the next state to transition to. This is a function of the inputs
+	 * and the current state of this FSM. This method should not have any side
+	 * effects on outputs. In other words, this method should only read or get
+	 * values to decide what state to go to.
+	 * @param input Global TeleopInput if robot in teleop mode or null if
+	 *        the robot is in autonomous mode.
+	 * @return FSM state for the next iteration
+	 */
+	private EveryBotIntakeFSMState nextState(TeleopInput input) {
+		if (input == null) {
+			return EveryBotIntakeFSMState.IDLE_STOP;
+		}
+		switch (currentState) {
+			case INTAKING:
+				if (needsReset && isMotorAllowed && toggleUpdate) {
+					timer.reset();
+					timer.start();
+					needsReset = false;
+				}
+				if (timer.hasElapsed(TIME_RESET_CURRENT)) {
+					currLogs[tick % AVERAGE_SIZE] = spinnerMotor.getOutputCurrent();
+					tick++;
+
+					double avgcone = 0;
+					for (int i = 0; i < AVERAGE_SIZE; i++) {
+						avgcone += currLogs[i];
+					}
+					avgcone /= AVERAGE_SIZE;
+					if (avgcone > CURRENT_THRESHOLD || !(input.isIntakeButtonPressed())) {
+						return EveryBotIntakeFSMState.IDLE_STOP;
+					}
+				}
+				return EveryBotIntakeFSMState.INTAKING;
+			case IDLE_STOP:
+				if (input.isOuttakeButtonPressed()
+					&& flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+					return EveryBotIntakeFSMState.OUTTAKING;
+				} else if (input.isIntakeButtonPressed()
+					&& !(flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD)) {
+					return EveryBotIntakeFSMState.INTAKING;
+				} else if (input.isFlipButtonPressed()) {
+					//&& arm.getEncoderCount() > BASE_THRESHOLD) {
+					if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD) {
+						return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+					} else {
+						return EveryBotIntakeFSMState.IDLE_FLIPCLOCKWISE;
+					}
+				} else {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				}
+			case IDLE_FLIPCLOCKWISE:
+				if (flipMotor.getEncoder().getPosition() > FLIP_THRESHOLD
+					|| input.isFlipAbortButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_STOP;
+				} else if (input.isFlipButtonPressed()) {
+					return EveryBotIntakeFSMState.IDLE_FLIPCOUNTERCLOCKWISE;
+				} else if (flipMotor.getEncoder().getPosition() < FLIP_THRESHOLD) {
```

Comment:
change last 'else if' to 'else' to ensure that we don't go to the next case

Reply:
Added